catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/converter/mgmp/MgmpConverter.java

```diff
@@ -0,0 +1,874 @@
+import org.codice.alliance.catalog.core.api.types.Security;
+import org.codice.alliance.catalog.transformer.mgmp.MgmpConstants;
+import org.codice.ddf.spatial.ogc.csw.catalog.common.GmdConstants;
+import org.codice.ddf.spatial.ogc.csw.catalog.converter.AbstractGmdConverter;
+import org.codice.ddf.spatial.ogc.csw.catalog.converter.XstreamPathValueTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thoughtworks.xstream.io.path.Path;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Associations;
+import ddf.catalog.data.types.Contact;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.DateTime;
+import ddf.catalog.data.types.Location;
+import ddf.catalog.data.types.Media;
+import ddf.catalog.data.types.Topic;
+
+/**
+ * Not thread safe.
+ */
+public class MgmpConverter extends AbstractGmdConverter {
+
+    // TODO i don't know how to output ISR_DATA_QUALITY, maybe deleted as per code review comments
+    @SuppressWarnings("unused")
+    public static final String ISR_DATA_QUALITY = "ext.isr.data-quality";
+
+    // TODO i don't know how to get the data from metacard to output ISR_DATA_QUALITY, maybe deleted as per code review comments
+    @SuppressWarnings("unused")
+    public static final List<String> DATA_QUALITY_LIST = Arrays.asList(
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/MGMP_UsabilityElement",
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/DQ_GriddedDataPositionalAccuracy",
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/DQ_RelativeInternalPositionalAccuracy",
+            "/MD_Metadata/dataQualityInfo/DQ_DataQuality/report/DQ_AbsoluteExternalPositionalAccuracy");
+
+    private static final String BOUNDING_BOX_FORMAT = "%.15f";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MgmpConverter.class);
+
+    private static final int MIN_CLOUD_COVERAGE = 0;
+
+    private static final int MAX_CLOUD_COVERAGE = 100;
+
+    private static final String ID_REGEX = "^.{32}$";
+
+    private static final int ID_SEGMENT_1_START = 0;
+
+    private static final int ID_SEGMENT_1_END = 8;
+
+    private static final int ID_SEGMENT_2_START = 8;
+
+    private static final int ID_SEGMENT_2_END = 12;
+
+    private static final int ID_SEGMENT_3_START = 12;
+
+    private static final int ID_SEGMENT_3_END = 16;
+
+    private static final int ID_SEGMENT_4_START = 16;
+
+    private static final int ID_SEGMENT_4_END = 20;
+
+    private static final int ID_SEGMENT_5_START = 20;
+
+    private static final int ID_SEGMENT_5_END = 32;
+
+    private static final String ID_SEPARATOR = "-";
+
+    private Predicate<String> crsFilter = s -> s.matches("^[^:]+:[^:]+$");
```

Comment:
You'll probably want to use a compiled `Pattern` for this since you're using it as a filter.